### ISSUE TITLE
Core X/Y Support

### ISSUE THREAD

**Issue by TinHead** [KevinOConnor/klipper#1]
Hello Kevin,

I'd like to give Klipper a try on my core xy build but I have trouble understanding the movement code which I would need to adapt on the python side.

Is there any chance I could get a quick example to get me on the right track?

Thank you,
Razvan

**Comment by KevinOConnor** [comment]
There have been several requests for corexy in the last couple of weeks.  I uploaded a preliminary development branch "work-corexy-20170306" for corexy.  The code is currently very rough and likely wont work great.  If you're feeling brave enough to test, then please provide feedback on the results.  (Be sure to select "kinematic: corexy" in the config.)  For the first few tests, be ready to issue an M112 (emergency stop) or pull the power if things don't go well.

**Comment by TinHead** [comment]
Perfect! I'll check it out and get back to you.

Thanks,
Razvan

**Comment by TinHead** [comment]
Oookeeey so I did install the firmware on my olde Arduino Mega 1280, figured out baud rate is 115200 on that one and fired up the daemon. 

I hit the error below ... not sure where to look for oid:

INFO:root:Starting Klippy...
INFO:root:Git version: 'v0.3.0-100-gb07bdd1'
INFO:root:CPU: 4 core Intel(R) Atom(TM) CPU D525   @ 1.80GHz
INFO:root:Building C code module
INFO:root:Start printer at Mon Mar 13 20:26:56 2017 (1489429616.5 387921.5)
INFO:root:===== Config file =====
INFO:root:[stepper_x]
INFO:root:step_pin = ar54
INFO:root:dir_pin = ar55
INFO:root:enable_pin = !ar38
INFO:root:step_distance = .01
INFO:root:endstop_pin = ^ar2
INFO:root:homing_speed = 50.0
INFO:root:position_min = 0
INFO:root:position_endstop = 0
INFO:root:position_max = 200
INFO:root:
INFO:root:[stepper_y]
INFO:root:step_pin = ar60
INFO:root:dir_pin = ar61
INFO:root:enable_pin = !ar56
INFO:root:step_distance = .01
INFO:root:endstop_pin = ^ar15
INFO:root:homing_speed = 50.0
INFO:root:position_min = 0
INFO:root:position_endstop = 0
INFO:root:position_max = 200
INFO:root:
INFO:root:[stepper_z]
INFO:root:step_pin = ar46
INFO:root:dir_pin = ar48
INFO:root:enable_pin = !ar62
INFO:root:step_distance = .01
INFO:root:endstop_pin = ^ar19
INFO:root:position_min = 0.1
INFO:root:position_endstop = 0.5
INFO:root:position_max = 200
INFO:root:
INFO:root:[extruder]
INFO:root:step_pin = ar26
INFO:root:dir_pin = ar28
INFO:root:enable_pin = !ar24
INFO:root:step_distance = .0022
INFO:root:nozzle_diameter = 0.400
INFO:root:filament_diameter = 1.750
INFO:root:heater_pin = ar10
INFO:root:sensor_type = ATC Semitec 104GT-2
INFO:root:sensor_pin = analog13
INFO:root:control = pid
INFO:root:pid_kp = 22.2
INFO:root:pid_ki = 1.08
INFO:root:pid_kd = 114
INFO:root:min_temp = 0
INFO:root:max_temp = 250
INFO:root:
INFO:root:[heater_bed]
INFO:root:heater_pin = ar8
INFO:root:sensor_type = EPCOS 100K B57560G104F
INFO:root:sensor_pin = analog14
INFO:root:control = watermark
INFO:root:min_temp = 0
INFO:root:max_temp = 130
INFO:root:
INFO:root:[fan]
INFO:root:pin = ar9
INFO:root:
INFO:root:[mcu]
INFO:root:serial = /dev/ttyUSB0
INFO:root:baud = 115200
INFO:root:pin_map = arduino
INFO:root:
INFO:root:[printer]
INFO:root:kinematics = corexy
INFO:root:max_velocity = 300
INFO:root:max_accel = 3000
INFO:root:max_z_velocity = 250
INFO:root:max_z_accel = 30
INFO:root:
INFO:root:=======================
INFO:root:Starting serial connect
INFO:root:Stats 387921.6: gcodein=0 print_time=0.000 buffer_time=0.000 print_stall=0  mcu_task_avg=0.000000 mcu_task_stddev=0.000000
INFO:root:Loaded 47 commands (v0.3.0-100-gb07bdd1-20170313_185010-sabayon)
INFO:root:MCU config: SERIAL_BAUD=115200 ADC_MAX=1024 CLOCK_FREQ=16000000 MCU=atmega1280 STATS_SUMSQ_BASE=256
INFO:root:Stats 387922.6: gcodein=0 print_time=0.000 buffer_time=0.000 print_stall=0 bytes_write=347 bytes_read=1924 bytes_retransmit=9 bytes_invalid=0 send_seq=53 receive_seq=53 retransmit_seq=2 srtt=0.003 rttvar=0.001 rto=0.025 ready_bytes=0 stalled_bytes=0 est_clock=0.000 last_ack_time=387922.560 last_ack_clock=45094090487 mcu_task_avg=0.000000 mcu_task_stddev=0.000000
INFO:root:Sending printer configuration...
INFO:root:is_shutdown: Invalid oid type
INFO:root:Dumping send queue 48 messages
INFO:root:Sent 0 387921.743443 387921.743443 8: seq: 11, identify offset=0 count=40
INFO:root:Sent 1 387922.242651 387922.242651 8: seq: 1e, identify offset=0 count=40
INFO:root:Sent 2 387922.249491 387922.249491 8: seq: 1f, identify offset=40 count=40
INFO:root:Sent 3 387922.256799 387922.256799 8: seq: 10, identify offset=80 count=40
INFO:root:Sent 4 387922.263790 387922.263790 9: seq: 11, identify offset=120 count=40
INFO:root:Sent 5 387922.270805 387922.270805 9: seq: 12, identify offset=160 count=40
INFO:root:Sent 6 387922.277794 387922.277794 9: seq: 13, identify offset=200 count=40
INFO:root:Sent 7 387922.284780 387922.284780 9: seq: 14, identify offset=240 count=40
INFO:root:Sent 8 387922.291750 387922.291750 9: seq: 15, identify offset=280 count=40
INFO:root:Sent 9 387922.298747 387922.298747 9: seq: 16, identify offset=320 count=40
INFO:root:Sent 10 387922.305735 387922.305735 9: seq: 17, identify offset=360 count=40
INFO:root:Sent 11 387922.312743 387922.312743 9: seq: 18, identify offset=400 count=40
INFO:root:Sent 12 387922.319763 387922.319763 9: seq: 19, identify offset=440 count=40
INFO:root:Sent 13 387922.326769 387922.326769 9: seq: 1a, identify offset=480 count=40
INFO:root:Sent 14 387922.333743 387922.333743 9: seq: 1b, identify offset=520 count=40
INFO:root:Sent 15 387922.340740 387922.340740 9: seq: 1c, identify offset=560 count=40
INFO:root:Sent 16 387922.347813 387922.347813 9: seq: 1d, identify offset=600 count=40
INFO:root:Sent 17 387922.354747 387922.354747 9: seq: 1e, identify offset=640 count=40
INFO:root:Sent 18 387922.361780 387922.361780 9: seq: 1f, identify offset=680 count=40
INFO:root:Sent 19 387922.368743 387922.368743 9: seq: 10, identify offset=720 count=40
INFO:root:Sent 20 387922.375775 387922.375775 9: seq: 11, identify offset=760 count=40
INFO:root:Sent 21 387922.382787 387922.382787 9: seq: 12, identify offset=800 count=40
INFO:root:Sent 22 387922.389745 387922.389745 9: seq: 13, identify offset=840 count=40
INFO:root:Sent 23 387922.396750 387922.396750 9: seq: 14, identify offset=880 count=40
INFO:root:Sent 24 387922.403793 387922.403793 9: seq: 15, identify offset=920 count=40
INFO:root:Sent 25 387922.410772 387922.410772 9: seq: 16, identify offset=960 count=40
INFO:root:Sent 26 387922.417892 387922.417892 9: seq: 17, identify offset=1000 count=40
INFO:root:Sent 27 387922.424759 387922.424759 9: seq: 18, identify offset=1040 count=40
INFO:root:Sent 28 387922.431736 387922.431736 9: seq: 19, identify offset=1080 count=40
INFO:root:Sent 29 387922.438738 387922.438738 9: seq: 1a, identify offset=1120 count=40
INFO:root:Sent 30 387922.445757 387922.445757 9: seq: 1b, identify offset=1160 count=40
INFO:root:Sent 31 387922.452743 387922.452743 9: seq: 1c, identify offset=1200 count=40
INFO:root:Sent 32 387922.459742 387922.459742 9: seq: 1d, identify offset=1240 count=40
INFO:root:Sent 33 387922.466785 387922.466785 9: seq: 1e, identify offset=1280 count=40
INFO:root:Sent 34 387922.473779 387922.473779 9: seq: 1f, identify offset=1320 count=40
INFO:root:Sent 35 387922.480741 387922.480741 9: seq: 10, identify offset=1360 count=40
INFO:root:Sent 36 387922.484727 387922.484727 9: seq: 11, identify offset=1370 count=40
INFO:root:Sent 37 387922.500939 387922.500418 6: seq: 12, get_uptime
INFO:root:Sent 38 387922.556263 387922.555742 6: seq: 13, get_status
INFO:root:Sent 39 387922.556784 387922.555811 6: seq: 14, get_config
INFO:root:Sent 40 387922.607865 387922.607258 7: seq: 15, allocate_oids count=16
INFO:root:Sent 41 387922.609341 387922.607545 17: seq: 16, config_soft_pwm_out oid=0 pin=12 cycle_ticks=1600000 default_value=0 max_duration=80000000
INFO:root:Sent 42 387922.610035 387922.607672 8: seq: 17, config_analog_in oid=1 pin=85
INFO:root:Sent 43 387922.610904 387922.607870 10: seq: 18, config_digital_out oid=3 pin=2 default_value=1 max_duration=0
INFO:root:Sent 44 387922.612032 387922.608114 13: seq: 19, config_soft_pwm_out oid=4 pin=62 cycle_ticks=160000 default_value=0 max_duration=0
INFO:root:Sent 45 387922.613160 387922.608298 13: seq: 1a, config_digital_out oid=5 pin=61 default_value=0 max_duration=80000000
INFO:root:Sent 46 387922.613855 387922.608434 8: seq: 1b, config_analog_in oid=6 pin=86
INFO:root:Sent 47 387922.614723 387922.608629 10: seq: 1c, config_digital_out oid=8 pin=31 default_value=1 max_duration=0
INFO:root:Dumping receive queue 20 messages
INFO:root:Receive: 0 387922.389524 387922.382787 49: seq: 13, identify_response offset=800 data='\xc7\x97\xa0\xa1\x96\xa6\x8c\x97\x89\xb3\xd8\xde\xff\xa3\xc7\x8b\x84:\xcd\xed\x94\xf9\xab\xa8(\xad4U\x05\x86#R\xe8\x0e\x8e\x0b\x9f\xfc\xe8\xd5'
INFO:root:Receive: 1 387922.396523 387922.389745 49: seq: 14, identify_response offset=840 data='fD\xf7\xe1\xa9i\xe3Y\xd88\xec\x85\x84\xba4\x8d\xd01xx\xbcU\xefq\xab\x95m\x9fz\x17\xbec\x9c\xf0\x18r\xc7\xd5Vw'
INFO:root:Receive: 2 387922.403527 387922.396750 49: seq: 15, identify_response offset=880 data='\xd1XP\x86\xd6\x1e\xbeH%U\x14\x9d\\d\xab 5\x8fu\x12\x92\n-\xce\xa6\xa7Bt\xb00\xad\x8a<K\xe2\x1aK}\x12T'
INFO:root:Receive: 3 387922.410523 387922.403793 49: seq: 16, identify_response offset=920 data="x\x8c\xdb-T4\xbe\xc6\xf2\x994*\x15\x96\x82\xc9\xd2\xc0\xd1\xbd'\x06\x92\xe7\x93\xf6\xd9\x188x\x1f\x9b2\rF\x7f\x0c\x86g."
INFO:root:Receive: 4 387922.417539 387922.410772 49: seq: 17, identify_response offset=960 data='\xd2z\n\x03>PP4\x82\x81\xb5\x1e\xe2+\xc6\x93\xd4\xa5\xd9\xd4\xddN\r>\x80\x91\xb9\x9c\xd6\xf9*-\xf3|\x11\ro\x063z'
INFO:root:Receive: 5 387922.424526 387922.417892 49: seq: 18, identify_response offset=1000 data='_\x96\xb1\xb3\xbd5$;\x9b\xf8\x17S\x94\xd8\x08\r\x96\x06\xed\xd8f\xf9\xca\xf7\xa2\x04\xabH\xc7$\x95y:3>+4_\xe94'
INFO:root:Receive: 6 387922.431518 387922.424759 49: seq: 19, identify_response offset=1040 data="\x02\xb3GCl\xf5\xc4\x0c+\xf8\xa4.\x9e\xea\xca'\x0b\xa3\xc8\xe3\x88\xc1\x83\xd0\x17\x1f\xc49\xee\x81\xd7\x08,&\n\xbe\x8a;Li"
INFO:root:Receive: 7 387922.438522 387922.431736 49: seq: 1a, identify_response offset=1080 data='\x0c]\xafE8Hb\x0f\xb7\xb69?\xf1\xe7DA\x99\x88\xf3\x9dM.\x9e\x02\x15\xf8.)(;\xda\x8a\xffFR\xce\xbe=\\\x1a'
INFO:root:Receive: 8 387922.445528 387922.438738 49: seq: 1b, identify_response offset=1120 data='\x8bU:\x14\x1f\rD\x9f?j\xfd"I\t\x1c\xb2\xa1\n\xc8A:\x91\xe8\xfa&\x0eS\x83N\xa2,\xe4\x18\xce\x86d:\xd0w\x88'
INFO:root:Receive: 9 387922.452522 387922.445757 49: seq: 1c, identify_response offset=1160 data="CI\xb9\xf1g\x05\xdb\xe8fC\xa3'\xa9\x842\x8f\x1c\x87A\x0b\x14c\x1b\x1e\xf6C\xc4\x15i9[\x98\x80\xbe\x02\x82%\xe5\xb4P"
INFO:root:Receive: 10 387922.459527 387922.452743 49: seq: 1d, identify_response offset=1200 data='\x81\xee\xd0\x89\xd0\x95\x10\xa1c\xb3\x13\x9e\xa2~(\xbc\xe9\x06\xaa\x86\xc7Y\rz(\r\x98\xd1\xc3\xf9\x1f_(\xa1\x9f\x00\x03\x8dh!'
INFO:root:Receive: 11 387922.466521 387922.459742 49: seq: 1e, identify_response offset=1240 data='\xc2\x08\xdce\xcf\x8ag\xd9)\xcf\xb2\xd3\xe6.;\xbb\xabk~\x9ag\xfc,+x!\xf8j\x91\xf1\xec\xd4\xca;\xb97\x9d\xbf\xbaR'
INFO:root:Receive: 12 387922.473518 387922.466785 49: seq: 1f, identify_response offset=1280 data='\xe3\xe1e\xe3\xe6\xe5\xf5\xab\x8b\xd7\xe2\xa7\x8b\xb7/\x90\x88\xf3E\x9ee\x88\x81;\x9d\xb8\xba\xf8\x83\x84\x19\xf48\xde\xed.\xdf\x927\xae'
INFO:root:Receive: 13 387922.480521 387922.473779 49: seq: 10, identify_response offset=1320 data='U\x8d\xe4\xf9\x8ap\x97\xaf\x7f\xbb\xfcE\xfc|\xfd\xf2\rA\x97\x19\xfd\xd1\xfd\xf2\xf6\xe2\xf6F\xdc\xbc\xbd\xbay\x03\xfc7/q=_'
INFO:root:Receive: 14 387922.484524 387922.480741 19: seq: 11, identify_response offset=1360 data=',g\x9f>\xfd\x03f\xf2J\xfa'
INFO:root:Receive: 15 387922.488524 387922.484727 9: seq: 12, identify_response offset=1370 data=''
INFO:root:Receive: 16 387922.503525 387922.500939 12: seq: 13, uptime high=10 clock=2143533111
INFO:root:Receive: 17 387922.559525 387922.556263 12: seq: 14, status clock=2144417527 status=1
INFO:root:Receive: 18 387922.560546 387922.556784 9: seq: 15, config is_config=0 crc=0 move_count=0
INFO:root:Receive: 19 387922.610534 387922.607865 7: seq: 16, is_shutdown static_string_id=9
ERROR:root:MCU error during connect
Traceback (most recent call last):
  File "klippy/klippy.py", line 172, in connect
    self.mcu.connect()
  File "/home/razvan/klipper/klippy/mcu.py", line 441, in connect
    self._send_config()
  File "/home/razvan/klipper/klippy/mcu.py", line 532, in _send_config
    self._shutdown_msg,))
error: Firmware error during config: Invalid oid type

I think it might be related to the mega 1280 vs 2650 I'm waiting on one of those...

**Comment by KevinOConnor** [comment]
On Mon, Mar 13, 2017 at 11:32:34AM -0700, TinHead wrote:
> Oookeeey so I did install the firmware on my olde Arduino Mega 1280, figured out baud rate is 115200 on that one and fired up the daemon. 
> 
> I hit the error below ... not sure where to look for oid:
[...]
> INFO:root:Receive: 19 387922.610534 387922.607865 7: seq: 16, is_shutdown static_string_id=9
> ERROR:root:MCU error during connect
> Traceback (most recent call last):
>   File "klippy/klippy.py", line 172, in connect
>     self.mcu.connect()
>   File "/home/razvan/klipper/klippy/mcu.py", line 441, in connect
>     self._send_config()
>   File "/home/razvan/klipper/klippy/mcu.py", line 532, in _send_config
>     self._shutdown_msg,))
> error: Firmware error during config: Invalid oid type

Thanks, there was an ordering problem with the initial config messages
which should be fixed now.  Can you pull the branch again and retry?

> I think it might be related to the mega 1280 vs 2650 I'm waiting on one of those...

There's no gain to the 2560 vs the 1280 - indeed the 1280 is probably
faster.

-Kevin

**Comment by TinHead** [comment]
OK still oid error: 

`python klippy/klippy.py config/example-corexy.cfg 
INFO:root:Starting Klippy...
INFO:root:Git version: 'v0.3.0-102-g150e51e-dirty'
INFO:root:CPU: 4 core Intel(R) Atom(TM) CPU D525   @ 1.80GHz
INFO:root:Start printer at Mon Mar 13 21:05:24 2017 (1489431924.6 390229.7)
INFO:root:===== Config file =====
INFO:root:[stepper_x]
INFO:root:step_pin = ar54
INFO:root:dir_pin = ar55
INFO:root:enable_pin = !ar38
INFO:root:step_distance = .01
INFO:root:endstop_pin = ^ar2
INFO:root:homing_speed = 50.0
INFO:root:position_min = 0
INFO:root:position_endstop = 0
INFO:root:position_max = 200
INFO:root:
INFO:root:[stepper_y]
INFO:root:step_pin = ar60
INFO:root:dir_pin = ar61
INFO:root:enable_pin = !ar56
INFO:root:step_distance = .01
INFO:root:endstop_pin = ^ar15
INFO:root:homing_speed = 50.0
INFO:root:position_min = 0
INFO:root:position_endstop = 0
INFO:root:position_max = 200
INFO:root:
INFO:root:[stepper_z]
INFO:root:step_pin = ar46
INFO:root:dir_pin = ar48
INFO:root:enable_pin = !ar62
INFO:root:step_distance = .01
INFO:root:endstop_pin = ^ar19
INFO:root:position_min = 0.1
INFO:root:position_endstop = 0.5
INFO:root:position_max = 200
INFO:root:
INFO:root:[extruder]
INFO:root:step_pin = ar26
INFO:root:dir_pin = ar28
INFO:root:enable_pin = !ar24
INFO:root:step_distance = .0022
INFO:root:nozzle_diameter = 0.400
INFO:root:filament_diameter = 1.750
INFO:root:heater_pin = ar10
INFO:root:sensor_type = ATC Semitec 104GT-2
INFO:root:sensor_pin = analog13
INFO:root:control = pid
INFO:root:pid_kp = 22.2
INFO:root:pid_ki = 1.08
INFO:root:pid_kd = 114
INFO:root:min_temp = 0
INFO:root:max_temp = 250
INFO:root:
INFO:root:[heater_bed]
INFO:root:heater_pin = ar8
INFO:root:sensor_type = EPCOS 100K B57560G104F
INFO:root:sensor_pin = analog14
INFO:root:control = watermark
INFO:root:min_temp = 0
INFO:root:max_temp = 130
INFO:root:
INFO:root:[fan]
INFO:root:pin = ar9
INFO:root:
INFO:root:[mcu]
INFO:root:serial = /dev/ttyUSB0
INFO:root:baud = 115200
INFO:root:pin_map = arduino
INFO:root:
INFO:root:[printer]
INFO:root:kinematics = corexy
INFO:root:max_velocity = 300
INFO:root:max_accel = 3000
INFO:root:max_z_velocity = 250
INFO:root:max_z_accel = 30
INFO:root:
INFO:root:=======================
INFO:root:Starting serial connect
INFO:root:Stats 390229.7: gcodein=0 print_time=0.000 buffer_time=0.000 print_stall=0  mcu_task_avg=0.000000 mcu_task_stddev=0.000000
INFO:root:Loaded 47 commands (v0.3.0-100-gb07bdd1-20170313_185010-sabayon)
INFO:root:MCU config: SERIAL_BAUD=115200 ADC_MAX=1024 CLOCK_FREQ=16000000 MCU=atmega1280 STATS_SUMSQ_BASE=256
INFO:root:Stats 390230.7: gcodein=0 print_time=0.000 buffer_time=0.000 print_stall=0 bytes_write=347 bytes_read=1924 bytes_retransmit=9 bytes_invalid=0 send_seq=43 receive_seq=43 retransmit_seq=2 srtt=0.003 rttvar=0.001 rto=0.025 ready_bytes=0 stalled_bytes=0 est_clock=0.000 last_ack_time=390230.691 last_ack_clock=82024978421 mcu_task_avg=0.000000 mcu_task_stddev=0.000000
INFO:root:Sending printer configuration...
INFO:root:is_shutdown: Invalid oid type
INFO:root:Dumping send queue 47 messages
INFO:root:Sent 0 390229.872689 390229.872689 8: seq: 11, identify offset=0 count=40
INFO:root:Sent 1 390230.371777 390230.371777 8: seq: 14, identify offset=0 count=40
INFO:root:Sent 2 390230.378950 390230.378950 8: seq: 15, identify offset=40 count=40
INFO:root:Sent 3 390230.385888 390230.385888 8: seq: 16, identify offset=80 count=40
INFO:root:Sent 4 390230.392857 390230.392857 9: seq: 17, identify offset=120 count=40
INFO:root:Sent 5 390230.399845 390230.399845 9: seq: 18, identify offset=160 count=40
INFO:root:Sent 6 390230.406839 390230.406839 9: seq: 19, identify offset=200 count=40
INFO:root:Sent 7 390230.413948 390230.413948 9: seq: 1a, identify offset=240 count=40
INFO:root:Sent 8 390230.420848 390230.420848 9: seq: 1b, identify offset=280 count=40
INFO:root:Sent 9 390230.427834 390230.427834 9: seq: 1c, identify offset=320 count=40
INFO:root:Sent 10 390230.434834 390230.434834 9: seq: 1d, identify offset=360 count=40
INFO:root:Sent 11 390230.441845 390230.441845 9: seq: 1e, identify offset=400 count=40
INFO:root:Sent 12 390230.448887 390230.448887 9: seq: 1f, identify offset=440 count=40
INFO:root:Sent 13 390230.455888 390230.455888 9: seq: 10, identify offset=480 count=40
INFO:root:Sent 14 390230.462859 390230.462859 9: seq: 11, identify offset=520 count=40
INFO:root:Sent 15 390230.469849 390230.469849 9: seq: 12, identify offset=560 count=40
INFO:root:Sent 16 390230.476851 390230.476851 9: seq: 13, identify offset=600 count=40
INFO:root:Sent 17 390230.483839 390230.483839 9: seq: 14, identify offset=640 count=40
INFO:root:Sent 18 390230.490848 390230.490848 9: seq: 15, identify offset=680 count=40
INFO:root:Sent 19 390230.497892 390230.497892 9: seq: 16, identify offset=720 count=40
INFO:root:Sent 20 390230.504845 390230.504845 9: seq: 17, identify offset=760 count=40
INFO:root:Sent 21 390230.511841 390230.511841 9: seq: 18, identify offset=800 count=40
INFO:root:Sent 22 390230.518881 390230.518881 9: seq: 19, identify offset=840 count=40
INFO:root:Sent 23 390230.525926 390230.525926 9: seq: 1a, identify offset=880 count=40
INFO:root:Sent 24 390230.532926 390230.532926 9: seq: 1b, identify offset=920 count=40
INFO:root:Sent 25 390230.539909 390230.539909 9: seq: 1c, identify offset=960 count=40
INFO:root:Sent 26 390230.546866 390230.546866 9: seq: 1d, identify offset=1000 count=40
INFO:root:Sent 27 390230.553838 390230.553838 9: seq: 1e, identify offset=1040 count=40
INFO:root:Sent 28 390230.560837 390230.560837 9: seq: 1f, identify offset=1080 count=40
INFO:root:Sent 29 390230.567904 390230.567904 9: seq: 10, identify offset=1120 count=40
INFO:root:Sent 30 390230.574884 390230.574884 9: seq: 11, identify offset=1160 count=40
INFO:root:Sent 31 390230.581849 390230.581849 9: seq: 12, identify offset=1200 count=40
INFO:root:Sent 32 390230.588847 390230.588847 9: seq: 13, identify offset=1240 count=40
INFO:root:Sent 33 390230.595899 390230.595899 9: seq: 14, identify offset=1280 count=40
INFO:root:Sent 34 390230.602847 390230.602847 9: seq: 15, identify offset=1320 count=40
INFO:root:Sent 35 390230.609841 390230.609841 9: seq: 16, identify offset=1360 count=40
INFO:root:Sent 36 390230.613826 390230.613826 9: seq: 17, identify offset=1370 count=40
INFO:root:Sent 37 390230.632404 390230.631883 6: seq: 18, get_uptime
INFO:root:Sent 38 390230.687516 390230.686995 6: seq: 19, get_status
INFO:root:Sent 39 390230.688037 390230.687077 6: seq: 1a, get_config
INFO:root:Sent 40 390230.739019 390230.738411 7: seq: 1b, allocate_oids count=16
INFO:root:Sent 41 390230.740494 390230.738697 17: seq: 1c, config_soft_pwm_out oid=0 pin=12 cycle_ticks=1600000 default_value=0 max_duration=80000000
INFO:root:Sent 42 390230.741189 390230.738847 8: seq: 1d, config_analog_in oid=1 pin=85
INFO:root:Sent 43 390230.742057 390230.739049 10: seq: 1e, config_digital_out oid=3 pin=2 default_value=1 max_duration=0
INFO:root:Sent 44 390230.743185 390230.739252 13: seq: 1f, config_soft_pwm_out oid=4 pin=62 cycle_ticks=160000 default_value=0 max_duration=0
INFO:root:Sent 45 390230.744314 390230.739432 13: seq: 10, config_digital_out oid=5 pin=61 default_value=0 max_duration=80000000
INFO:root:Sent 46 390230.745008 390230.739570 8: seq: 11, config_analog_in oid=6 pin=86
INFO:root:Dumping receive queue 20 messages
INFO:root:Receive: 0 390230.518615 390230.511841 49: seq: 19, identify_response offset=800 data='\xc7\x97\xa0\xa1\x96\xa6\x8c\x97\x89\xb3\xd8\xde\xff\xa3\xc7\x8b\x84:\xcd\xed\x94\xf9\xab\xa8(\xad4U\x05\x86#R\xe8\x0e\x8e\x0b\x9f\xfc\xe8\xd5'
INFO:root:Receive: 1 390230.525620 390230.518881 49: seq: 1a, identify_response offset=840 data='fD\xf7\xe1\xa9i\xe3Y\xd88\xec\x85\x84\xba4\x8d\xd01xx\xbcU\xefq\xab\x95m\x9fz\x17\xbec\x9c\xf0\x18r\xc7\xd5Vw'
INFO:root:Receive: 2 390230.532616 390230.525926 49: seq: 1b, identify_response offset=880 data='\xd1XP\x86\xd6\x1e\xbeH%U\x14\x9d\\d\xab 5\x8fu\x12\x92\n-\xce\xa6\xa7Bt\xb00\xad\x8a<K\xe2\x1aK}\x12T'
INFO:root:Receive: 3 390230.539628 390230.532926 49: seq: 1c, identify_response offset=920 data="x\x8c\xdb-T4\xbe\xc6\xf2\x994*\x15\x96\x82\xc9\xd2\xc0\xd1\xbd'\x06\x92\xe7\x93\xf6\xd9\x188x\x1f\x9b2\rF\x7f\x0c\x86g."
INFO:root:Receive: 4 390230.546621 390230.539909 49: seq: 1d, identify_response offset=960 data='\xd2z\n\x03>PP4\x82\x81\xb5\x1e\xe2+\xc6\x93\xd4\xa5\xd9\xd4\xddN\r>\x80\x91\xb9\x9c\xd6\xf9*-\xf3|\x11\ro\x063z'
INFO:root:Receive: 5 390230.553613 390230.546866 49: seq: 1e, identify_response offset=1000 data='_\x96\xb1\xb3\xbd5$;\x9b\xf8\x17S\x94\xd8\x08\r\x96\x06\xed\xd8f\xf9\xca\xf7\xa2\x04\xabH\xc7$\x95y:3>+4_\xe94'
INFO:root:Receive: 6 390230.560617 390230.553838 49: seq: 1f, identify_response offset=1040 data="\x02\xb3GCl\xf5\xc4\x0c+\xf8\xa4.\x9e\xea\xca'\x0b\xa3\xc8\xe3\x88\xc1\x83\xd0\x17\x1f\xc49\xee\x81\xd7\x08,&\n\xbe\x8a;Li"
INFO:root:Receive: 7 390230.567625 390230.560837 49: seq: 10, identify_response offset=1080 data='\x0c]\xafE8Hb\x0f\xb7\xb69?\xf1\xe7DA\x99\x88\xf3\x9dM.\x9e\x02\x15\xf8.)(;\xda\x8a\xffFR\xce\xbe=\\\x1a'
INFO:root:Receive: 8 390230.574617 390230.567904 49: seq: 11, identify_response offset=1120 data='\x8bU:\x14\x1f\rD\x9f?j\xfd"I\t\x1c\xb2\xa1\n\xc8A:\x91\xe8\xfa&\x0eS\x83N\xa2,\xe4\x18\xce\x86d:\xd0w\x88'
INFO:root:Receive: 9 390230.581621 390230.574884 49: seq: 12, identify_response offset=1160 data="CI\xb9\xf1g\x05\xdb\xe8fC\xa3'\xa9\x842\x8f\x1c\x87A\x0b\x14c\x1b\x1e\xf6C\xc4\x15i9[\x98\x80\xbe\x02\x82%\xe5\xb4P"
INFO:root:Receive: 10 390230.588615 390230.581849 49: seq: 13, identify_response offset=1200 data='\x81\xee\xd0\x89\xd0\x95\x10\xa1c\xb3\x13\x9e\xa2~(\xbc\xe9\x06\xaa\x86\xc7Y\rz(\r\x98\xd1\xc3\xf9\x1f_(\xa1\x9f\x00\x03\x8dh!'
INFO:root:Receive: 11 390230.595624 390230.588847 49: seq: 14, identify_response offset=1240 data='\xc2\x08\xdce\xcf\x8ag\xd9)\xcf\xb2\xd3\xe6.;\xbb\xabk~\x9ag\xfc,+x!\xf8j\x91\xf1\xec\xd4\xca;\xb97\x9d\xbf\xbaR'
INFO:root:Receive: 12 390230.602620 390230.595899 49: seq: 15, identify_response offset=1280 data='\xe3\xe1e\xe3\xe6\xe5\xf5\xab\x8b\xd7\xe2\xa7\x8b\xb7/\x90\x88\xf3E\x9ee\x88\x81;\x9d\xb8\xba\xf8\x83\x84\x19\xf48\xde\xed.\xdf\x927\xae'
INFO:root:Receive: 13 390230.609613 390230.602847 49: seq: 16, identify_response offset=1320 data='U\x8d\xe4\xf9\x8ap\x97\xaf\x7f\xbb\xfcE\xfc|\xfd\xf2\rA\x97\x19\xfd\xd1\xfd\xf2\xf6\xe2\xf6F\xdc\xbc\xbd\xbay\x03\xfc7/q=_'
INFO:root:Receive: 14 390230.613618 390230.609841 19: seq: 17, identify_response offset=1360 data=',g\x9f>\xfd\x03f\xf2J\xfa'
INFO:root:Receive: 15 390230.617616 390230.613826 9: seq: 18, identify_response offset=1370 data=''
INFO:root:Receive: 16 390230.635620 390230.632404 12: seq: 19, uptime high=19 clock=419717909
INFO:root:Receive: 17 390230.690622 390230.687516 12: seq: 1a, status clock=420599797 status=1
INFO:root:Receive: 18 390230.691615 390230.688037 9: seq: 1b, config is_config=0 crc=0 move_count=0
INFO:root:Receive: 19 390230.741633 390230.739019 7: seq: 1c, is_shutdown static_string_id=9
ERROR:root:MCU error during connect
Traceback (most recent call last):
  File "klippy/klippy.py", line 172, in connect
    self.mcu.connect()
  File "/home/razvan/klipper/klippy/mcu.py", line 444, in connect
    self._send_config()
  File "/home/razvan/klipper/klippy/mcu.py", line 535, in _send_config
    self._shutdown_msg,))
error: Firmware error during config: Invalid oid type
`
looks like it does not like analog14 if I read this correct ...

**Comment by KevinOConnor** [comment]
On Mon, Mar 13, 2017 at 12:07:46PM -0700, TinHead wrote:
> OK still oid error: 
[...]
> INFO:root:Receive: 16 390230.635620 390230.632404 12: seq: 19, uptime high=19 clock=419717909
> INFO:root:Receive: 17 390230.690622 390230.687516 12: seq: 1a, status clock=420599797 status=1
> INFO:root:Receive: 18 390230.691615 390230.688037 9: seq: 1b, config is_config=0 crc=0 move_count=0
> INFO:root:Receive: 19 390230.741633 390230.739019 7: seq: 1c, is_shutdown static_string_id=9
> ERROR:root:MCU error during connect
> Traceback (most recent call last):
>   File "klippy/klippy.py", line 172, in connect
>     self.mcu.connect()
>   File "/home/razvan/klipper/klippy/mcu.py", line 444, in connect
>     self._send_config()
>   File "/home/razvan/klipper/klippy/mcu.py", line 535, in _send_config
>     self._shutdown_msg,))
> error: Firmware error during config: Invalid oid type

Can you confirm you ran FIRMWARE_RESTART after updating the host
software?  At first glance, it looks like the firmware was still in an
error state from the last attempt.

-Kevin

**Comment by TinHead** [comment]
OK you got me on that :) 
Reset of the Mega did it :)

**Comment by TinHead** [comment]
I can confirm comms work fine now got octoprint talking to the daemon and daemon to the mega.

Commands work fine too.

Going to test movement next I have to fix some wiring first.

**Comment by TinHead** [comment]
Quick question: does klipper support dual steppers on z with separate drivers?

**Comment by KevinOConnor** [comment]
On Mon, Mar 13, 2017 at 03:17:37PM -0700, TinHead wrote:
> Quick question: does klipper support dual steppers on z with separate drivers?

Klipper doesn't currently have support for multiple extruders.  I
don't think it would be difficult to add, but I don't have the
hardware and there doesn't seem to be a standard for multiple
extruders.

-Kevin

**Comment by TinHead** [comment]
I did not mean multiple extruders, I mean dual stepper motors for an axis driven at the same time using separate drivers, sort of port mirroring.

**Comment by KevinOConnor** [comment]
On Mon, Mar 13, 2017 at 03:35:54PM -0700, TinHead wrote:
> I did not mean multiple extruders, I mean dual stepper motors for an axis driven at the same time using separate drivers, sort of port mirroring.

Ah, sorry.  It's not currently supported, but it is not difficult to
add.  Out of curiosity though, why use two drivers instead of just
wiring both motors to a single driver?

-Kevin

**Comment by TinHead** [comment]
You get to control the current per motor. Also both motors and driver run cooler then when wired in parallel.

**Comment by TinHead** [comment]
Update: 
I'm blocked by hardware for now, once I get that solved I should be able to do some more testing.

**Comment by MrJulz** [comment]
I can confirm the CoreXY kinematics are working for me.

**Comment by TinHead** [comment]
Hi, 

Finally figured out my hardware issues, got x/y moving and homing. One thing I have noticed is that the following sequence fails atm:

1. Home x/y
2. Move x/y to random location
3. Home again => results in hitting x switch and bail with firmare error 

To get it back to work I have to issue clear_shutdown..

I still have to get Z going so that axis is not homed in this case, could it be related?

I'll get back with more once I get the Z going.

**Comment by KevinOConnor** [comment]
On Sun, Mar 19, 2017 at 01:47:02AM -0700, TinHead wrote:
> Hi, 
> 
> Finally figured out my hardware issues, got x/y moving and homing. One thing I have noticed is that the following sequence fails atm:
> 
> 1. Home x/y
> 2. Move x/y to random location
> 3. Home again => results in hitting x switch and bail with firmare error 
> 
> To get it back to work I have to issue clear_shutdown..
> 
> I still have to get Z going so that axis is not homed in this case, could it be related?
> 
> I'll get back with more once I get the Z going.

What version are you on?  There were several fixes and your
description above sounds like one of the initial bugs that were fixed
(latest is 1e3f245f).

-Kevin

**Comment by TinHead** [comment]
Ok seems I have missed some commits there gonna pull and let you know ....

**Comment by TinHead** [comment]
Issue still happens, I'm on latest:
`git pull
remote: Counting objects: 64, done.
remote: Total 64 (delta 33), reused 33 (delta 33), pack-reused 31
Unpacking objects: 100% (64/64), done.
From https://github.com/KevinOConnor/klipper
   150e51e..1e3f245  work-corexy-20170306 -> origin/work-corexy-20170306
   0f2478b..5ff2d5a  master               -> origin/master
Updating 150e51e..1e3f245
Fast-forward
 klippy/corexy.py      |  6 +++-
 klippy/gcode.py       | 86 ++++++++++++++++++++++++++++++++++-----------------
 klippy/klippy.py      |  2 ++
 klippy/mcu.py         |  6 ++--
 klippy/serialhdl.py   |  2 +-
 klippy/stepcompress.c |  8 +++--
 klippy/stepper.py     | 14 ++++++---
 klippy/toolhead.py    |  9 ++++--
 8 files changed, 90 insertions(+), 43 deletions(-)
`

Here is the terminal from Octoprint:
`Recv: !! Must home axis first: 10.000 0.000 0.000 [0.000]
Recv: ok
Recv: ok
Send: G91
Send: G28 X0 Y0
Recv: ok
Send: G90
Send: M105
Recv: ok
Recv: ok
Recv: ok T:23.5 /0.0 B:23.2 /0.0
Send: M105
Recv: ok T:22.8 /0.0 B:23.6 /0.0
Send: G91
Send: G1 X10 F6000
Recv: ok
Send: G90
Recv: ok
Recv: ok
Send: G91
Send: G1 X10 F6000
Send: G90
Recv: ok
Recv: ok
Recv: ok
Send: G91
Send: G1 X10 F6000
Recv: ok
Send: G90
Recv: ok
Recv: ok
Send: G91
Send: G1 X10 F6000
Recv: ok
Send: G90
Recv: ok
Recv: ok
Send: G91
Send: G1 X10 F6000
Send: G90
Recv: ok
Recv: ok
Recv: ok
Send: M105
Recv: ok T:23.4 /0.0 B:23.4 /0.0
Send: G91
Send: G1 Y10 F6000
Send: G90
Recv: ok
Recv: ok
Recv: ok
Send: G91
Send: G1 Y10 F6000
Send: G90
Recv: ok
Recv: ok
Recv: ok
Send: G91
Send: G1 Y10 F6000
Send: G90
Recv: ok
Recv: ok
Recv: ok
Send: G91
Send: G1 Y10 F6000
Recv: ok
Send: G90
Recv: ok
Recv: ok
Send: M105
Recv: ok T:23.1 /0.0 B:23.5 /0.0
Send: G91
Send: G28 X0 Y0
Recv: ok
Send: G90
Recv: !! Internal error on command:"G28"
Recv: ok
Recv: ok
`
And debug from klippy:

`
Sent 91 395316.557901 395316.557380 6: seq: 11, get_status
Sent 92 395317.558840 395317.558319 6: seq: 12, get_status
Sent 93 395317.629758 395317.629150 7: seq: 13, stepper_get_position oid=7
Sent 94 395317.680322 395317.679715 7: seq: 14, stepper_get_position oid=10
Sent 95 395317.736291 395317.730996 61: seq: 15, set_next_step_dir oid=7 dir=1, set_next_step_dir oid=10 dir=0, queue_step oid=7 interval=576846453 count=1 add=0, queue_step oid=7 interval=16656 count=2 add=-4766, queue_step oid=10 interval=576846453 count=1 add=0, queue_step oid=10 interval=16656 count=2 add=-4766, queue_step oid=7 interval=9076 count=5 add=-780, queue_step oid=10 interval=9076 count=5 add=-780
Sent 96 395317.741586 395317.731160 61: seq: 16, queue_step oid=7 interval=5623 count=10 add=-201, queue_step oid=10 interval=5623 count=10 add=-201, queue_step oid=7 interval=3739 count=13 add=-48, queue_step oid=10 interval=3739 count=13 add=-48, queue_step oid=7 interval=3334 count=598 add=0, queue_step oid=10 interval=3334 count=598 add=0, queue_step oid=7 interval=3333 count=949 add=0, queue_step oid=10 interval=3333 count=949 add=0
Sent 97 395317.746882 395317.731492 61: seq: 17, queue_step oid=7 interval=3480 count=12 add=111, queue_step oid=10 interval=3480 count=12 add=111, queue_step oid=7 interval=5187 count=6 add=369, queue_step oid=10 interval=5187 count=6 add=369, queue_step oid=7 interval=8109 count=3 add=1715, queue_step oid=10 interval=8109 count=3 add=1715, queue_step oid=7 interval=17007 count=1 add=0, queue_step oid=10 interval=17007 count=1 add=0
Sent 98 395317.752264 395317.732469 62: seq: 18, end_stop_home oid=9 clock=582316692 rest_ticks=6666 pin_value=0, set_next_step_dir oid=7 dir=0, queue_step oid=7 interval=45938 count=2 add=-28782, queue_step oid=7 interval=11260 count=3 add=-1500, queue_step oid=7 interval=6916 count=9 add=-53, queue_step oid=7 interval=6667 count=1154 add=0, queue_step oid=7 interval=6666 count=601 add=0, queue_step oid=7 interval=6667 count=1202 add=0
Sent 99 395317.755649 395317.737069 39: seq: 19, emergency_stop, queue_step oid=7 interval=6666 count=224 add=0, set_next_step_dir oid=10 dir=1, queue_step oid=10 interval=3712673698 count=1 add=0, queue_step oid=7 interval=7245 count=3 add=1147, queue_step oid=7 interval=11390 count=2 add=5766
Dumping receive queue 20 messages
Receive: 0 395315.896561 395315.706616 14: seq: 11, analog_in_state oid=1 next_clock=548290760 value=7847
Receive: 1 395315.947560 395315.706616 14: seq: 11, analog_in_state oid=6 next_clock=549090760 value=7856
Receive: 2 395316.198567 395315.706616 14: seq: 11, analog_in_state oid=1 next_clock=553090760 value=7849
Receive: 3 395316.249566 395315.706616 14: seq: 11, analog_in_state oid=6 next_clock=553890760 value=7852
Receive: 4 395316.500577 395315.706616 14: seq: 11, analog_in_state oid=1 next_clock=557890760 value=7851
Receive: 5 395316.551566 395315.706616 14: seq: 11, analog_in_state oid=6 next_clock=558690760 value=7851
Receive: 6 395316.561557 395316.557901 12: seq: 12, status clock=554273081 status=0
Receive: 7 395316.802567 395316.557901 14: seq: 12, analog_in_state oid=1 next_clock=562690760 value=7853
Receive: 8 395316.853568 395316.557901 14: seq: 12, analog_in_state oid=6 next_clock=563490760 value=7850
Receive: 9 395316.922566 395316.557901 18: seq: 12, stats count=121227 sum=80000541 sumsq=225348066
Receive: 10 395317.104565 395316.557901 14: seq: 12, analog_in_state oid=1 next_clock=567490760 value=7856
Receive: 11 395317.156568 395316.557901 14: seq: 12, analog_in_state oid=6 next_clock=568290760 value=7848
Receive: 12 395317.406567 395316.557901 14: seq: 12, analog_in_state oid=1 next_clock=572290760 value=7856
Receive: 13 395317.458567 395316.557901 14: seq: 12, analog_in_state oid=6 next_clock=573090760 value=7848
Receive: 14 395317.562557 395317.558840 12: seq: 13, status clock=570171896 status=0
Receive: 15 395317.579564 395317.558840 9: seq: 13, end_stop_state oid=9 homing=0 pin=0
Receive: 16 395317.633556 395317.629758 9: seq: 14, stepper_position oid=7 pos=6430
Receive: 17 395317.684551 395317.680322 9: seq: 15, stepper_position oid=10 pos=6395
Receive: 18 395317.709561 395317.680322 14: seq: 15, analog_in_state oid=1 next_clock=577090760 value=7853
Receive: 19 395317.761552 395317.755649 7: seq: 1a, shutdown static_string_id=18
Dumping gcode input 43 blocks
Read 395284.561876: 'M105\n'
Read 395289.561463: 'M105\n'
Read 395294.562636: 'M105\n'
Read 395294.824805: 'G91\n'
Read 395294.829720: 'G1 X10 F6000\n'
Read 395294.832716: 'G90\n'
Read 395298.718560: 'G91\n'
Read 395298.726039: 'G28 X0 Y0\n'
Read 395298.739581: 'G90\n'
Read 395302.471996: 'M105\n'
Read 395304.564502: 'M105\n'
Read 395305.090316: 'G91\n'
Read 395305.092706: 'G1 X10 F6000\n'
Read 395305.094596: 'G90\n'
Read 395305.981880: 'G91\n'
Read 395305.982590: 'G1 X10 F6000\n'
Read 395305.983991: 'G90\n'
Read 395306.814470: 'G91\n'
Read 395306.818418: 'G1 X10 F6000\n'
Read 395306.820259: 'G90\n'
Read 395307.562189: 'G91\n'
Read 395307.568688: 'G1 X10 F6000\n'
Read 395307.571273: 'G90\n'
Read 395308.361218: 'G91\n'
Read 395308.368412: 'G1 X10 F6000\n'
Read 395308.371651: 'G90\n'
Read 395309.565106: 'M105\n'
Read 395310.550810: 'G91\n'
Read 395310.558436: 'G1 Y10 F6000\n'
Read 395310.559735: 'G90\n'
Read 395311.327508: 'G91\n'
Read 395311.328364: 'G1 Y10 F6000\n'
Read 395311.331484: 'G90\n'
Read 395312.109319: 'G91\n'
Read 395312.110176: 'G1 Y10 F6000\n'
Read 395312.116138: 'G90\n'
Read 395312.917586: 'G91\n'
Read 395312.919320: 'G1 Y10 F6000\n'
Read 395312.922268: 'G90\n'
Read 395314.566786: 'M105\n'
Read 395315.608195: 'G91\n'
Read 395315.611707: 'G28 X0 Y0\n'
Read 395315.622809: 'G90\n'
`

**Comment by KevinOConnor** [comment]
On Mon, Mar 20, 2017 at 02:20:07AM -0700, TinHead wrote:
> Issue still happens, I'm on latest:

Thanks for testing.  Can you attach the full klipper.log file?  If
it's large, can you gzip it before attaching.

-Kevin

**Comment by TinHead** [comment]
You are welcome I'll test further, got tired of the mainstream firmwares:)

Unfortunatelly I cannot provide the full log right now I will tomorrow, hope it's still there ... Else I'll replicate it again. I don't think it is a switch issue...

**Comment by TinHead** [comment]
OK back with an update:
- I can no longer replicate homing issue so I guess it was an issue on my part
- everything seems to work I actually managed to print some tests -- looking good :+1: 
- sometimes I've seen errors: Recv: // Firmware shutdown: adc out of range however not sure what to correlate with ... did not happen during my test prints so not sure ... will get back to you if I figure it out 

All in all things are pretty smooth I'll stick with Klipper at least for now :)

**Comment by TinHead** [comment]
I think we can close... maybe merge to master?

**Comment by KevinOConnor** [comment]
Lets leave this one open until it is merged.  I need to rebase the changes and I'll likely ask you to retest with the merged version.  I hope to get to that in a few days.

The "adc out of range" indicates the firmware detected one of the thermistors (either bed or extruder) had a temperature outside of the config's min_temp/max_temp.  I recommend you investigate that.

**Comment by TinHead** [comment]
OK Let me know I'll test and get back to you, if any new issue is found I'll open a new ticket once in master I hate handling multiple issues in the same tread :)

**Comment by KevinOConnor** [comment]
On Thu, Mar 30, 2017 at 10:08:14AM -0700, TinHead wrote:
> OK Let me know I'll test and get back to you, if any new issue is
> found I'll open a new ticket once in master I hate handling multiple
> issues in the same tread :)

Thanks.  Can you pull the corexy branch again and verify everything
still works?  This is rebase'd to master (so you'll need to use "git
fetch; git checkout work-corexy-20170306").

Merging with master pulled in some important bug fixes, so you'll want
to use this code regardless.

Once I have confirmation on the recent changes I'll commit the corexy
code into master.

-Kevin

**Comment by KevinOConnor** [comment]
I received that basic functionality works with the merged corexy, so I've committed this to master.  Thanks for testing.
